src/Nest/Domain/ICustomJson.cs

```diff
@@ -2,14 +2,12 @@
 
 namespace Nest
 {
-	//If an object implements this then it can handle its own json representation
+	// If an object implements this then it can handle its own json representation
 	public interface ICustomJson
 	{
 		object GetCustomJson();
 	}
 
-	public interface IDomainObject
-	{
-		
-	}
+	// Any object that implements this will automatically be added to the serialization contract
+	public interface INestSerializable { }
```

Comment:
lets make this a proper summary: `Any object that implements this interface will automatically have all the JsonProperties of all its implementing interfaces discovered.`